### PR TITLE
Added pagination.

### PR DESCRIPTION
**Pull request by iayti**
I added pagination support using @jbogard's Contoso University sample. I also added detailed response for pagination. I see the #121 @MCCshreyas commit and implement @jasontaylordev  advice. Thank you for your interest. 

Have a nice coding 😃

```csharp
PaginatedList<TodoItemDto> list = await _context.TodoItems
                .Where(x => x.ListId == request.ListId)
                .OrderBy(x => x.Title)
                .ProjectTo<TodoItemDto>(_mapper.ConfigurationProvider)
                .PaginatedListAsync(request.PageNumber, request.PageSize);
````

Pagination_Sample_Request_Response

### CODE REVIEW

**Review by vksantoshsunny — APPROVED** [review]

File: src/Application/Common/Models/PaginationQuery.cs [hunk 1]
```diff
@@ -0,0 +1,8 @@
+﻿namespace CleanArchitecture.Application.Common.Models
+{
+    public class PaginationQuery
+    {
+        public int PageNumber { get; set; } = 1;
+        public int PageSize { get; set; } = 10;
+    }
+}
```
Comment (jasontaylordev) [inline src/Application/Common/Models/PaginationQuery.cs:8]: Recommend dropping pagination query, I'd rather just add the properties directly to the query. That way query can remain isolated from one another and they remain explicit with regards to the input requirements.

File: src/Application/Common/Mappings/MappingExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading.Tasks;
+
+    using Microsoft.EntityFrameworkCore;
+
+    using AutoMapper;
+    using AutoMapper.QueryableExtensions;
+    
+    using Models;
```
Comment (jasontaylordev) [inline src/Application/Common/Mappings/MappingExtensions.cs:12]: Using statements outside namespaces, sorted alphabetically, without line breaks between

File: src/Application/Common/Models/PaginationResponse.cs [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+        public List<T> Items { get; set; }
+
+        public int PageIndex { get; set; }
+
+        public int TotalPages { get; set; }
+
+        public int TotalCount { get; set; }
+
+    }
+}
```
Comment (jasontaylordev) [inline src/Application/Common/Models/PaginationResponse.cs:18]: This is not required, the PaginationList<T> has the complete details.

**Review by jasontaylordev — CHANGES_REQUESTED** [review]
Thanks a lot for the PR, it's looking good so far - please can you make the requested changes?

**Comment by iayti** [comment]
> Thanks a lot for the PR, it's looking good so far - please can you make the requested changes?

Thank you for the advice, it became simpler and more beautiful this way. 😃

**Pull request merged**